WebRoot/js/zimbraMail/briefcase/view/ZmPreviewPaneView.js

```diff
@@ -372,19 +372,28 @@ ZmPreviewPaneView.prototype._listSelectionListener = function(ev, item) {
         return;
     }
 
-    if(ev.field == ZmItem.F_EXPAND && this._detailListView._isExpandable(item)){
-        this._detailListView.expandItem(item);   
-    } else if(this._controller.isReadingPaneOn() && item){
-    	if (ev.kbNavEvent) {
-    		if (this._listSelectionShortcutDelayActionId) {
-    			AjxTimedAction.cancelAction(this._listSelectionShortcutDelayActionId); 
-    		}
-    		this._delayedSelectionItem = item;
-    		this._listSelectionShortcutDelayActionId = AjxTimedAction.scheduleAction(this._listSelectionShortcutDelayAction,
-    				ZmPreviewPaneView.LIST_SELECTION_SHORTCUT_DELAY)
-    	} else {
-    		this._previewView.set(item);
-    	}
+    if (ev.field == ZmItem.F_EXPAND && this._detailListView._isExpandable(item)) {
+        this._detailListView.expandItem(item);
+    } else if (this._controller.isReadingPaneOn() && item) {
+        if (ev.kbNavEvent) {
+            if (this._listSelectionShortcutDelayActionId) {
+                AjxTimedAction.cancelAction(this._listSelectionShortcutDelayActionId);
+            }
+            this._delayedSelectionItem = item;
+            this._listSelectionShortcutDelayActionId = AjxTimedAction.scheduleAction(
+                this._listSelectionShortcutDelayAction,
+                ZmPreviewPaneView.LIST_SELECTION_SHORTCUT_DELAY
+            );
+        } else {
+            this._previewView.set(item);
+        }
+    } else if (!this._controller.isReadingPaneOn() && item && item.folderId == ZmFolder.ID_FILE_SHARED_WITH_ME) {
+        var restUrl = item.getRestUrl();
+        restUrl = AjxStringUtil.fixCrossDomainReference(restUrl);
+        var req = new XMLHttpRequest();
+        req.open("GET", restUrl, true);
+        req.onreadystatechange = this._handleNoPreviewUrlResponse.bind(this, req, item);
+        req.send();
```

Comment:
entire condition can be optimised like below
} else if {item} {
      if (this._controller.isReadingPaneOn()) {
            ....
      } else if (item.folderId == ZmFolder.ID_FILE_SHARED_WITH_ME) {
            ....
      }
}

Reply:
done